### PR TITLE
VSF2-177 Reorganization of category tree structure

### PR DESCRIPTION
**Pull request by klloz**

### CODE REVIEW

File: src/Doctrine/Repository/TaxonRepository.php [hunk 1]
```diff
@@ -10,24 +10,90 @@
...
+use Doctrine\ORM\EntityRepository;
 use Doctrine\ORM\QueryBuilder;
-use Sylius\Bundle\TaxonomyBundle\Doctrine\ORM\TaxonRepository as BaseTaxonRepository;
+use Gedmo\Tree\Entity\Repository\NestedTreeRepository;
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\ResourceRepositoryTrait;
 use Sylius\Component\Taxonomy\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface as BaseTaxonRepositoryInterface;
 
-final class TaxonRepository extends BaseTaxonRepository implements TaxonRepositoryInterface
+final class TaxonRepository extends NestedTreeRepository implements TaxonRepositoryInterface
```
Comment (senghe) [inline src/Doctrine/Repository/TaxonRepository.php:20]: Logic in NestedTreeRepository is too complex for merging it without tests. Could you write PHPUnit integration tests for the repository?

**Review by senghe — CHANGES_REQUESTED** [review]

File: src/Doctrine/Repository/TaxonRepository.php (hunk 1 above)
Comment (klloz) [reply src/Doctrine/Repository/TaxonRepository.php:20]: I wrote a test for a new method. The rest of them is either a call or a copy of the original ones.

File: tests/Integration/Doctrine/Repository/TaxonRepositoryTest.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+namespace Tests\BitBag\SyliusVueStorefront2Plugin\Integration\Doctrine\Repository;
+
+use BitBag\SyliusVueStorefront2Plugin\Doctrine\Repository\TaxonRepository;
+use Faker\Factory;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Taxonomy\Model\TaxonTranslation;
+
+final class TaxonRepositoryTest extends AbstractRepositoryTest
+{
+    public function testCreatesChildrenByChannelMenuTaxonQueryBuilder(): void
```
Comment (senghe) [inline tests/Integration/Doctrine/Repository/TaxonRepositoryTest.php:20]: I would use the lchrusciel ApiTestCase library to define the better fixtures, and then do better matches than checking count of taxons

**Review by senghe — CHANGES_REQUESTED** [review]

**Comment by senghe** [comment]
The build has failed due to one broken spec. I'm merging this to develop, as it is needed by frontend. I'll fix the spec directly on develop.

**Pull request merged**